### ISSUE TITLE
It is posible to type at the begging of the 7ZWS

### ISSUE THREAD

**Issue by pjasiun** [ckeditor/ckeditor5-engine#613]
This one is tricky.

First of all because of: https://github.com/ckeditor/ckeditor5-engine/issues/462 it is not possible to reproduce it easily. On the other hand when https://github.com/ckeditor/ckeditor5-engine/issues/462 will be fixed this could be another reason of many issues.

Then problem is that:

We use 7 ZWS as an inline filling character. And we assume that user will always type at the end of it so only check if the string starts with 7 ZWS. But for some strange reason it is possible to put the caret at the begging of the 7 ZWS.

To reproduce the bug you can modify the classic editor manual test adding:
```js
editor.document.schema.registerItem( 't', '$block' );

buildModelConverter().for( editor.data.modelToView, editor.editing.modelToView )	
	.fromElement( 't' )
	.toElement( new ViewContainerElement( 'span', { contenteditable: false } ) );

buildViewConverter().for( editor.data.viewToModel )
	.fromElement( 'span' )
	.toElement( 't' );

editor.setData( '<p><span>foo</span></p><p>bar</p>' );
```

Then:
 - init editor,
 - put the caret at the beginning of the second line (before `bar`),
 - press arrow up,
 - start typing,
 - check console.

You will see that you type at the beginning of the inline filter what is unexpected. Then when you move the selection you get:
```
renderer.js:279 Uncaught CKEditorError: view-renderer-render-no-inline-filler: No inline filler on expected position.
```
Because renderer checks if the text starts with 7 ZWS to thread it as the filler and since you added text at the beginning of it, it will not work. Also, other algorithms may not work, like the one to jump over the filler.

The question is how it is possible to type before the 7 ZWS and if we can fix it to type after. If it is not possible some algorithms may need a fix to handle the situation where 7 ZWS are after the text.

**Comment by Reinmar** [comment]
@f1ames does this ticket make any sense now?

**Comment by f1ames** [comment]
@Reinmar makes sense, we should definitely take a look on this issue closely when working on inline filler improvements mentioned in #1342.

**Comment by f1ames** [comment]
I tried to reproduce this issue (on `http://localhost:8125/ckeditor5-core/tests/manual/article.html`) with the following code:

```js
model.schema.register( 't', { inheritAllFrom: '$block' } );

editor.conversion.for( 'downcast' ).add( downcastElementToElement( {
	model: 't',
	view: ( modelItem, viewWriter ) => viewWriter.createEditableElement( 'span', { contenteditable: false } )
} ) );

editor.conversion.for( 'upcast' ).add( upcastElementToElement( {
	model: 't',
	view: 'span'
} ) );

editor.setData( '<p><span>foo</span></p><p>bar</p>' );
```

but typed text was inserted after filler and not before (though I'm not sure my repro steps are correct):

![zws2](https://user-images.githubusercontent.com/1061942/39769187-c9469a1e-52eb-11e8-9780-2e22be3d4f22.gif)

@pjasiun does it look to you that this issue was solved along the way or you are still able to reproduce it?
